### PR TITLE
OpInfo porting for logsumexp operator

### LINKED ISSUE #56132: Unexpected error when passing integer tensor to logsumexp
Passing an integer tensor to logsumexp:
```
t = torch.tensor([1,2,3])
torch.logsumexp(t, 0)
```
Leads to an error message : 
```
RuntimeError: value cannot be converted to type int64_t without overflow: inf
```

However this is not the expected behavior. It would be helpful to investigate this issue.

### PR DESCRIPTION
**Pull request by iramazanli**
Fixes #{issue number}

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -3897,6 +3912,16 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            supports_inplace_autograd=True,
            safe_casts_outputs=True,
            sample_inputs_func=sample_inputs_xlogy),
+    OpInfo('logsumexp',
+           dtypes=floating_types_and(torch.bfloat16),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3916]: it also supports float16 on CUDA

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -3897,6 +3912,16 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+           supports_out=False,
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3917]: logsumexp does support out

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -3897,6 +3912,16 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+           supports_complex_autograd=False,
+           skips=(
+               # logsumexp get error for the following test.
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3920]: can you elaborate on this error?

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -648,6 +648,21 @@ def sample_inputs_xlogy(self, device, dtype, requires_grad, **kwargs):
         ),
     )
 
+def sample_inputs_logsumexp(self, device, dtype, requires_grad):
+    inputs = [
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:652]: prefer tuples to lists if the data structure doesn't have to be mutable

File: torch/testing/_internal/common_methods_invocations.py [hunk 5]
```diff
@@ -648,6 +648,21 @@ def sample_inputs_xlogy(self, device, dtype, requires_grad, **kwargs):
...
+        ((), (0,), True),
+        ((S, S), (1,), True)
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:654]: Let's duplicate this case but with keepdim=False, too

File: torch/testing/_internal/common_methods_invocations.py (hunk 3 above)
Comment (iramazanli) [reply torch/testing/_internal/common_methods_invocations.py:3920]: yes and the error message is as following for the given test : 

Failure in testing nodes' autodifferentiation. One or more nodes were not expected to be autodiffed but were found in a DifferentiableGraph or in a FusionGroup of a DifferentiableGraph. Did you intend for these nodes to be autodiffed? If so, change this test to expect autodifferentiation. 
Specifically:
  ['aten::logsumexp'] were not expected to be in one of the DifferentiableGraphs but were.

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3947]: I see. You'll need to play around with the autodiff metadata, see here:

https://github.com/pytorch/pytorch/blob/facbcec2980783a011a046643577592480754210/torch/testing/_internal/common_methods_invocations.py#L150

Based on the error message, I think the change this needs is

```
assert_autodiffed=True
```

File: torch/testing/_internal/common_methods_invocations.py [hunk 6]
```diff
@@ -3923,6 +3939,16 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            supports_inplace_autograd=True,
            safe_casts_outputs=True,
            sample_inputs_func=sample_inputs_xlogy),
+    OpInfo('logsumexp',
+           dtypes=floating_types_and(torch.bfloat16),
+           dtypesIfCUDA=all_types_and(torch.bfloat16),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3944]: As mentioned above, it looks like the op also supported torch.float16 (AKA torch.half) on CUDA

Note that torch.bfloat16 is not torch.float16

**Comment by mruberry** [comment]
What's going on with the CUDA integer input failures?

**Comment by iramazanli** [comment]
> What's going on with the CUDA integer input failures?

I get the following error : RuntimeError: value cannot be converted to type int64_t without overflow: inf

File: torch/testing/_internal/common_methods_invocations.py [hunk 7]
```diff
@@ -4196,6 +4212,12 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            supports_inplace_autograd=True,
            safe_casts_outputs=True,
            sample_inputs_func=sample_inputs_xlogy),
+    OpInfo('logsumexp',
+           dtypes=floating_types_and(torch.bfloat16),
+           dtypesIfCUDA=all_types_and(torch.bfloat16, torch.half, torch.int64),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:4217]: Shouldn't the integer dtypes be removed for CUDA?

Would you file an issue for the odd error when passing integer tensors to logsumexp along with a snippet to reproduce it, too?

**Review by mruberry — APPROVED** [review]
Cool!

**Pull request closed without merging**